projects/ngx-sub-form/src/lib/ngx-sub-form.component.spec.ts

```diff
@@ -328,6 +328,70 @@ describe(`NgxSubFormComponent`, () => {
       }, 0);
     });
   });
+
+  describe('getFormGroupControlOptions', () => {
+    interface Numbered {
+      numberOne: number;
+      numberTwo: number;
+    }
+
+    class ValidatedSubComponent extends NgxSubFormComponent<Numbered> {
+      protected getFormControls() {
+        return {
+          numberOne: new FormControl(null, Validators.required),
+          numberTwo: new FormControl(null, Validators.required),
+        };
+      }
+
+      public getFormGroupControlOptions(): AbstractControlOptions {
+        return {
+          validators: [
+            formGroup => {
+              const sum = formGroup.value.numberOne + formGroup.value.numberTwo;
```

Comment:
I wonder if we shouldn't pass a cb to `getFormGroupControlOptions` that'd return the `formGroupValues`.
This way it'd be type safe.

Reply:
Yea I thought about that, but it kinda feels like a bit of a risk that validation is not on what is being passed to the validator function. We could possibly achieve typesafety by abusing the `AbstractControlOptions` type?

Reply:
Ok I managed the above so it is fully type safe, without having to pass a callback

Reply:
Nice one! :fire: